src/kbmod/work_unit.py

```diff
@@ -0,0 +1,182 @@
+
+
+class WorkUnit:
+    """The work unit is a storage and I/O class for all of the data
+    needed for a full run of KBMOD, including the: the search parameters,
+    data files, and the data provenance metadata.
+    """
+
+    def __init__(self, im_stack=None, config=None):
+        self.im_stack = im_stack
+        self.config = config
+
+    @classmethod
+    def from_fits(cls, filename):
+        """Create a WorkUnit from a single FITS file.
+
+        A WorkUnit is written as a FITS file with the following extensions:
+            0 - Primary header with overall metadata
+            1 or "metadata" - The data provenance metadata
+            2 or "kbmod_config" - The search parameters
+            3+ - Image extensions for the science layer ("SCI_i"),
+                variance layer ("VAR_i"), mask layer ("MSK_i"), and
+                PSF ("PSF_i") of each image.
+
+        Parameters
+        ----------
+        filename : `str`
+            The file to load.
+
+        Returns
+        -------
+        result : `WorkUnit`
+            The loaded WorkUnit.
+        """
+        if not Path(filename).is_file():
+            raise ValueError(f"WorkUnit file {filename} not found.")
+
+        imgs = []
+        with fits.open(filename) as hdul:
+            num_layers = len(hdul)
+            if num_layers < 5:
+                raise ValueError(f"WorkUnit file has too few extensions {len(hdul)}.")
+
+            # TODO - Read in provenance metadata from extension #1.
+
+            # Read in the search parameters from the 'kbmod_config' extension.
+            config = SearchConfiguration.from_hdu(hdul["kbmod_config"])
+
+            # Read the size and order information from the primary header.
+            num_images = hdul[0].header["NUMIMG"]
+            if len(hdul) != 4 * num_images + 3:
+                raise ValueError(
+                    f"WorkUnit wrong number of extensions. Expected "
+                    f"{4 * num_images + 3}. Found {len(hdul)}."
+                )
+
+            # Read in all the image files.
+            for i in range(num_images):
+                # Read in science, variance, and mask layers.
+                sci = hdu_to_raw_image(hdul[f"SCI_{i}"])
+                var = hdu_to_raw_image(hdul[f"VAR_{i}"])
+                msk = hdu_to_raw_image(hdul[f"MSK_{i}"])
+
+                # Read the PSF layer.
+                p = PSF(hdul[f"PSF_{i}"].data)
+
+                imgs.append(LayeredImage(sci, var, msk, p))
+
+        im_stack = ImageStack(imgs)
+        return WorkUnit(im_stack=im_stack, config=config)
+
+    def to_fits(self, filename, overwrite=False):
+        """Write the WorkUnit to a single FITS file.
+
+        Uses the following extensions:
+            0 - Primary header with overall metadata
+            1 or "metadata" - The data provenance metadata
+            2 or "kbmod_config" - The search parameters
+            3+ - Image extensions for the science layer ("SCI_i"),
```

Comment:
Remember that the docstrings are markdown rendered into documentation by Sphinx. To render a proper list that discriminates between references and literals try:

```
The FITS file will have at least the following extensions:

0. ``PRIMARY`` extension
1. ``METADATA`` extension containing provenance 
2. ``KBMOD_CONFIG`` extension containing search parameters
3. (+) any additional image extensions are named ``SCI_i``, ``VAR_i``, ``MSK_i`` 
and ``PSF_i`` for the science, variance, mask and PSF of each image respectively, 
where ``i`` runs from 0 to number of images in the `WorkUnit`.
```